subsys/mgmt/hawkbit/hawkbit.c

```diff
@@ -1007,7 +1007,7 @@ static void response_cb(struct http_response *rsp, enum http_final_call final_da
 		} else {
 			hb_context->code_status = HAWKBIT_METADATA_ERROR;
 		}
-		return;
+		return -EACCES;
```

Comment:
this would lead to a hb_context->code_status being overwritten, as `http_client_req()` would now return a negative value in line 1139, maybe change this to 0 and I will follow up with a separate PR for hawkBit, to really make use this feature.

Reply:
ok, done